tests/model_serving/model_server/authentication/conftest.py

```diff
@@ -0,0 +1,302 @@
+import shlex
+
+import pytest
+from kubernetes.dynamic import DynamicClient
+from ocp_resources.inference_service import InferenceService
+from ocp_resources.namespace import Namespace
+from ocp_resources.resource import ResourceEditor
+from ocp_resources.role_binding import RoleBinding
+from ocp_resources.role import Role
+from ocp_resources.secret import Secret
+from ocp_resources.service_account import ServiceAccount
+from ocp_resources.serving_runtime import ServingRuntime
+from ocp_resources.authorino import Authorino
+from ocp_utilities.infra import get_pods_by_name_prefix
+from pyhelper_utils.shell import run_command
+
+
+from tests.model_serving.model_server.authentication.constants import (
+    CAIKIT_STR,
+    CAIKIT_TGIS_RUNTIME_STR,
+    CAIKIT_TGIS_SERVING_TEMPLATE_STR,
+    GRPC_STR,
+    HTTP_STR,
+    SERVERLESS_STR,
+)
+from tests.model_serving.model_server.authentication.utils import (
+    create_isvc_view_role,
+    get_s3_secret_dict,
+)
+from tests.model_serving.model_server.utils import create_isvc
+from utilities.serving_runtime import ServingRuntimeFromTemplate
+
+
+@pytest.fixture(scope="session")
+def skip_if_no_authorino_operator(admin_client: DynamicClient):
+    if not Authorino(
+        client=admin_client,
+        name="authorino",
+        namespace="redhat-ods-applications-auth-provider",
+    ).exists:
+        pytest.skip("Authorino operator is missing from the cluster")
+
+
+@pytest.fixture(scope="session")
+def models_s3_bucket_name(pytestconfig) -> str:
+    bucket_name = pytestconfig.option.models_s3_bucket_name
+    if not bucket_name:
+        raise ValueError(
+            "Models bucket name is not set. "
+            "Either pass with `--models-s3-bucket-name` or set `MODELS_S3_BUCKET_NAME` environment variable"
+        )
+
+    return bucket_name
+
+
+@pytest.fixture(scope="session")
+def models_s3_bucket_region(pytestconfig) -> str:
+    bucket_name = pytestconfig.option.models_s3_bucket_region
+    if not bucket_name:
+        raise ValueError(
+            "Models bucket region is not set. "
+            "Either pass with `--models-s3-bucket-region` or set `MODELS_S3_BUCKET_REGION` environment variable"
+        )
+
+    return bucket_name
+
+
+@pytest.fixture(scope="class")
+def s3_models_storage_uri(request, models_s3_bucket_name) -> str:
+    return f"s3://{models_s3_bucket_name}/{request.param['model-dir']}/"
+
+
+@pytest.fixture(scope="session")
+def models_s3_endpoint(models_s3_bucket_region) -> str:
+    return f"https://{models_s3_bucket_region}.amazonaws.com/"
+
+
+@pytest.fixture(scope="class")
+def endpoint_s3_secret(
```

Comment:
This is already defined here: https://github.com/opendatahub-io/opendatahub-tests/pull/22/files/38b1f728c8d82955805212d36cf8c47f03d17a10#diff-ba4079b730a33a115ca68082ee9cdafe3a1d4cbff3aba9d09d508deacafbb183R51

Reply:
+1

Reply:
because it requires the namespace

Reply:
Can it not be passed as an argument to the fixture?

Reply:
no

it is ok to have fixtures which do the same thing; this is why we have a function to do it in one place

the tests need to stay clean. separate resources for each test